kit/src/main/java/com/oracle/javafx/scenebuilder/kit/template/TemplatesBaseWindowController.java

```diff
@@ -84,28 +73,52 @@ protected void controllerDidCreateStage() {
     @Override
     protected void controllerDidLoadFxml() {
         super.controllerDidLoadFxml();
-        assert templateContainer != null;
 
-        basicDesktopApp.setUserData(Template.BASIC_DESKTOP_APP);
-        complexDesktopApp.setUserData(Template.COMPLEX_DESKTOP_APP);
-        emptyPhoneApp.setUserData(Template.EMPTY_PHONE_APP);
-        basicPhoneApp.setUserData(Template.BASIC_PHONE_APP);
+        initTemplates();
     }
 
-    protected void setupTemplateButtonHandlers() {
-        setupTemplateButtonHandlers(templateContainer);
-    }
+    /**
+     * Dynamically generates templates and adds them to fxml-pre-defined [desktopPane] and [mobilePane].
+     */
+    private void initTemplates() {
+        FlowPane desktopPane;
+        FlowPane mobilePane;
+
+        try {
+            ScrollPane templatesRoot = (ScrollPane) getRoot().lookup("#templatesRoot");
+            desktopPane = (FlowPane) templatesRoot.getContent().lookup("#desktopPane");
+            mobilePane = (FlowPane) templatesRoot.getContent().lookup("#mobilePane");
+        } catch (Exception e) {
+            System.out.println("Failed to lookup() [desktopPane] and [mobilePane]: ");

```

Comment:
We should use the logger for all printouts, and at least every new PR should use it.
We shouldn't print stack traces to the console either. This is only visible if user runs from command line. 
We should log the error and the exception to the logs, which are visible to anyone.

Reply:
Good idea, I'll fix these here and open an issue for others in the codebase.